src/share/classes/com/sun/btrace/client/Main.java

```diff
@@ -149,6 +152,12 @@ public static void main(String[] args) {
                     statsdDef = args[++count];
                 } else if (args[count].equals("-v")) {
                     // already processed
+                } else if (args[count].equals("-ip") && !ipDefined) {
+                    ip = args[++count];
+                    if (!ip.matches("((25[0-5]|2[0-4]\\d|((1\\d{2})|([1-9]?\\d)))\\.){3}(25[0-5]|2[0-4]\\d|((1\\d{2})|([1-9]?\\d)))")) {
```

Comment:
This will match IPs only - will not validate the default value of `localhost`. There should not be any validation as both host name and IP is accepted (unless you want to make a really complicated regex to validate)

Reply:
I am sorry (*>﹏<*)